testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_channel_revocation.py

```diff
@@ -0,0 +1,2059 @@
+# from datetime import datetime
+import random
+import pytest
+import time
+# from concurrent.futures import ThreadPoolExecutor
+from couchbase.bucket import Bucket
+from keywords import document
+from keywords.MobileRestClient import MobileRestClient
+from keywords.SyncGateway import sync_gateway_config_path_for_mode, create_sync_gateways
+from keywords.utils import add_new_fields_to_doc, host_for_url, log_info, add_additional_new_field_to_doc
+from libraries.testkit.cluster import Cluster
+
+from requests.exceptions import HTTPError
+
+DB1 = "db1"
+DB2 = "db2"
+
+
+@pytest.mark.listener
+@pytest.mark.channel_revocation
+@pytest.mark.parametrize("auto_purge_setting", [
+    pytest.param("enabled"),
+    pytest.param("default"),
+    pytest.param("disabled")
+])
+def test_auto_purge_setting_impact(params_from_base_test_setup, auto_purge_setting):
+    """
+        @summary:
+        Channel Access Revocation Test Plan (ISGR) #1
+        1. on passive SGW, create docs:
+                doc_A belongs to channel A only,
+                doc_AnB belongs to channel A and channel B
+        2. start a pull continous replication on active SGW with auto_purge_setting parameter
+        3. verify active SGW have pulled the doc_A and doc_AnB
+        4. revoke the user access to channel A
+        5. verify expected doc auto purge result on active SGW
+    """
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    mode = params_from_base_test_setup["mode"]
+    sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+
+    # check sync gateway version
+    if sync_gateway_version < "3.0.0":
+        pytest.skip('This test cannot run with Sync Gateway version below 3.0')
+
+    # prepare sync gateway environment
+    config = sync_gateway_config_path_for_mode("sync_gateway_sg_replicate", mode)
+
+    sg1, sg2 = create_sync_gateways(
```

Comment:
this is actually does not create sgws into two different SGW clusters. Both sgws connect to two buckets of server cluster . This becomes more of sg db to SG-db replication in same cluster rather than having two different clusters.
Can you look into the tests that I have written for ISGR2. You can use have tests in this folder, but you can use the logic to have as 2 differrent cluster.
Basically after you create sgws or resetting the cluster, you can redeploy the sgw config to have sgw1 connect to only data-bucket1 and sg2 to only data-bucket2 which will make it as 2 differrent clusters

Reply:
updated my code based on your comments